### ISSUE TITLE
As a user, I'd like to be able to close my browser window after I submit a web3 tx, so I can do other things without canceling.

### ISSUE THREAD

**Issue by owocki** [gitcoinco/web#128]
014429d8ce15a484e59e3ad7760be5475e455cb0

# the issue

Once a user submits a new bounty / claim bounty / accept bounty, their browser is what polls web3 for the tx to be mined.  When the tx is mined, their browser receives the updated info from the blockchain, and submits it to the server.

this is a problem because if the user closers browser window, or if they lose web3 connectivity, the backend won't update.

# workaround

a workaround is if an admin is using the `process_bounty.js` helper script to process the bounty.

# proposed root cause fix

there is a serverside management command called `sync_geth`.  it is partially implemented right now.  when it is fully implemented, it will poll / listen for changes to the jobs on web3 and update the server side immediately

**Comment by gitcoinbot** [comment]
__This issue now has a funding of 0.05 ETH (33.89 USD) attached to it.__

 * If you would like to work on this issue you can claim it [here](https://gitcoin.co/funding/details?url=https://github.com/gitcoinco/web/issues/128).
 * If you've completed this issue and want to claim the bounty you can do so [here](https://gitcoin.co/funding/details?url=https://github.com/gitcoinco/web/issues/128)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $10303.76 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by wingleung** [comment]
Where can I find the partially implemented `sync_get` command?

**Comment by owocki** [comment]
@wingleung https://github.com/gitcoinco/web/blob/master/app/dashboard/management/commands/sync_geth.py

**Comment by sc0Vu** [comment]
@owocki Maybe it's better to run sync_geth as daemon on serverside?

**Comment by owocki** [comment]
sure.  im open to that... we could put it into `supervisorctl` but first we need to get it working

**Comment by step21** [comment]
Is someone working on this? Just curious, as I think I am not that familiar with the codebase to decide which way to handle this.

**Comment by abitrolly** [comment]
Nobody claimed it, so feel free to do it. My question are - how state is stored and what is stored?

Is there something like this stored on server?

    change requested -> transaction pending -> transaction completed
                                            -> transaction cancelled

So that when you reload the page, the state is shown on issue page?

**Comment by step21** [comment]
You can look at /app/dashboard/models.py for example to get an idea of the database layout.
There is a model BountySyncRequest that gets then manually or otherwise processed as far as I can say.

**Comment by step21** [comment]
One question I have about this - I am probably missing something, but what about this is not working right?
As I understand it, sync_get and the corresponding parts like process_bounty_changes should work, but there would need to bet a method/part in new_bounty.js, process_bounty.js, claim_bounty.js (or maybe at least claim_bounty) that adds a BountySyncRequest, which will then be picked up by from there. Correct?

**Comment by owocki** [comment]
> sync_get

i believe that sync_get exits out with an exception right now.

also @jhaas is working on standardbounties integration.. we should hold off until that is merged in  https://github.com/gitcoinco/web/pull/140

**Comment by step21** [comment]
ok. then I will in the meantime try running without docker, cause running those managing commands in docker was a pain

**Comment by owocki** [comment]
we should make it easier to run management commands in docker.. @mbeacom might know whats standard here :)

**Comment by mbeacom** [comment]
@step21 @owocki 
You can execute management command via something like:

`docker-compose exec web python manage.py collectstatic`

The above example is broken down like so:
`docker-compose <action> <stack_service/container_name> <command>`

So you're telling docker-compose to `execute` the command: `python manage.py collectstatic` against the `web` service container.

So, for this particular situation, you could use:  `docker-compose exec web python manage.py sync_get`

Let me know if you have any further questions or concerns!

**Comment by owocki** [comment]
this is unblocked now that standardbounties is out the door!

**Comment by step21** [comment]
I will check if I can commit sth today

**Comment by step21** [comment]
So... should this be combined with those two referenced issues or can it be split up somehow?

**Comment by owocki** [comment]
im going to start on this tomorrow

**Comment by mbeacom** [comment]
@vs77bb This has been folded into https://github.com/gitcoinco/web/pull/379 - We have a tentative rollout set for tomorrow.

**Comment by owocki** [comment]
this is done! 🎉
